File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        unordered_map<int,int>mp;
        for(int i=0;i<n;i++){
            int req=target-nums[i];
            if(mp.find(req)!=mp.end()){
                return {mp[req],i};
            }
            mp[nums[i]]=i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> mp = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            int targetEle = target - nums[i];
            if (mp.containsKey(targetEle) && mp.get(targetEle) != i) {
                return new int[]{i, mp.get(targetEle)};
            }
            mp.put(nums[i], i);
        }
        
        return new int[]{-1, -1}; // No solution case
    }
}
